gateway/server.go

```diff
@@ -1799,16 +1799,31 @@ func handleDashboardRegistration(gw *Gateway) {
 	}()
 }
 
-func (gw *Gateway) startDRL() {
-	switch {
-	case gw.GetConfig().ManagementNode:
-		return
-	case gw.GetConfig().EnableSentinelRateLimiter, gw.GetConfig().EnableRedisRollingLimiter:
-		return
-	}
-	mainLog.Info("Initialising distributed rate limiter")
-	gw.setupDRL()
-	gw.startRateLimitNotifications()
+func (gw *Gateway) startDRL() bool {
+	gwConfig := gw.GetConfig()
+
+	disabled := gwConfig.ManagementNode || gwConfig.EnableSentinelRateLimiter || gw.GetConfig().EnableRedisRollingLimiter
+
+	gw.drlOnce.Do(func() {
+		drlManager := &drl.DRL{}
+		gw.DRLManager = drlManager
+
+		if disabled {
+			return
+		}
+
+		mainLog.Info("Initialising distributed rate limiter")
+
+		nodeID := gw.GetNodeID() + "|" + gw.hostDetails.Hostname
+
+		drlManager.ThisServerID = nodeID
+		drlManager.Init(gw.ctx)
+
+		log.Debug("DRL: Setting node ID: ", nodeID)
+
+		gw.startRateLimitNotifications()
+	})
+	return true
```

Comment:
`startDRL` is always returning `true`, don't we need a case to return false when it's disabled?

Reply:
there is no disabled according to L1854

Reply:
do we need to return bool in that case?